engines/asylum/resources/encounters.cpp

```diff
@@ -0,0 +1,1688 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "asylum/resources/encounters.h"
+
+#include "asylum/resources/actor.h"
+#include "asylum/resources/object.h"
+#include "asylum/resources/script.h"
+#include "asylum/resources/worldstats.h"
+
+#include "asylum/system/cursor.h"
+#include "asylum/system/graphics.h"
+#include "asylum/system/screen.h"
+#include "asylum/system/speech.h"
+#include "asylum/system/text.h"
+
+#include "asylum/views/scene.h"
+#include "asylum/views/video.h"
+
+#include "asylum/asylum.h"
+#include "asylum/respack.h"
+#include "asylum/staticres.h"
+
+namespace Asylum {
+
+#define KEYWORD_MASK 0xFFF
+
+#define OPCODE_NAME(index) (index > 25 ? "INVALID" : opcodeNames[index])
+const char *opcodeNames[] = {
+	"Return",
+	"SetScriptVariable",
+	"SetCounterFromVars",
+	"SetOffset",
+	"SetOffsetIfCounterNegative",
+	"SetOffsetIfCounterNegativeOrNull",
+	"SetOffsetIfCounterIsNull",
+	"SetOffsetIfCounterIsNotNull",
+	"SetOffsetIfCounterPositiveOrNull",
+	"SetOffsetIfCounterPositive",
+	"SetCurrentItemOptions",
+	"ClearCurrentItemOptions",
+	"SetItemOptions",
+	"CloseDialog",
+	"ResetSpeech",
+	"SetVariable",
+	"IncrementScriptVariable",
+	"ProcessVariable3",
+	"AddRemoveReactionHive",
+	"UNUSED (19)",
+	"UNUSED (20)",
+	"SetCounterFromActorReactions",
+	"UNUSED (22)",
+	"PrepareMovie",
+	"SetClearGameFlag",
+	"SetCounterFromGameFlag"
+};
+
+Common::String Encounter::ScriptEntry::toString() {
+	return Common::String::format("0x%02X: %s (%d, %d)", opcode, OPCODE_NAME(opcode), param1, param2);
+}
+
+Encounter::Encounter(AsylumEngine *engine) : _vm(engine),
+	_index(0), _speechResourceId(0), _item(NULL), _objectId1(kObjectNone), _objectId2(kObjectNone), _objectId3(kObjectNone),
```

Comment:
This line needs adding a default initializer for _soundResourceId i.e. Change to:
````
	_index(0), _speechResourceId(0), _soundResourceId(kResourceNone), _item(NULL), _objectId1(kObjectNone), _objectId2(kObjectNone), _objectId3(kObjectNone),
````

Reply:
This was identified by a run with valgrind as a condition depending on uninitialized value at line 460 of this file.